storm-core/src/jvm/org/apache/storm/daemon/supervisor/SupervisorData.java

```diff
@@ -0,0 +1,339 @@
+        return stormVersion;
+    }
+
+    public void setStormVersion(String stormVersion) {
+        this.stormVersion = stormVersion;
+    }
+
+    public ConcurrentHashMap<String, String> getWorkerThreadPidsAtom() {
+        return workerThreadPidsAtom;
+    }
+
+    public void setWorkerThreadPidsAtom(ConcurrentHashMap<String, String> workerThreadPidsAtom) {
+        this.workerThreadPidsAtom = workerThreadPidsAtom;
+    }
+
+    public IStormClusterState getStormClusterState() {
+        return stormClusterState;
+    }
+
+    public void setStormClusterState(IStormClusterState stormClusterState) {
+        this.stormClusterState = stormClusterState;
+    }
+
+    public LocalState getLocalState() {
+        return localState;
+    }
+
+    public void setLocalState(LocalState localState) {
+        this.localState = localState;
+    }
+
+    public String getSupervisorId() {
+        return supervisorId;
+    }
+
+    public void setSupervisorId(String supervisorId) {
+        this.supervisorId = supervisorId;
+    }
+
+    public String getAssignmentId() {
+        return assignmentId;
+    }
+
+    public void setAssignmentId(String assignmentId) {
+        this.assignmentId = assignmentId;
+    }
+
+    public String getHostName() {
+        return hostName;
+    }
+
+    public void setHostName(String hostName) {
+        this.hostName = hostName;
+    }
+
+    public ConcurrentHashMap<Long, LocalAssignment> getCurrAssignment() {
+        return currAssignment;
+    }
+
+    public void setCurrAssignment(Map<Long, LocalAssignment> currAssignment) {
+        this.currAssignment.clear();
+        this.currAssignment.putAll(currAssignment);
+    }
+
+    public StormTimer getHeartbeatTimer() {
+        return heartbeatTimer;
+    }
+
+    public void setHeartbeatTimer(StormTimer heartbeatTimer) {
+        this.heartbeatTimer = heartbeatTimer;
+    }
+
+    public StormTimer getEventTimer() {
+        return eventTimer;
+    }
+
+    public void setEventTimer(StormTimer eventTimer) {
+        this.eventTimer = eventTimer;
+    }
+
+    public StormTimer getBlobUpdateTimer() {
+        return blobUpdateTimer;
+    }
+
+    public void setBlobUpdateTimer(StormTimer blobUpdateTimer) {
+        this.blobUpdateTimer = blobUpdateTimer;
+    }
+
+    public Localizer getLocalizer() {
+        return localizer;
+    }
+
+    public void setLocalizer(Localizer localizer) {
+        this.localizer = localizer;
+    }
+
+    public AtomicInteger getSyncRetry() {
+        return syncRetry;
+    }
+
+    public void setSyncRetry(AtomicInteger syncRetry) {
+        this.syncRetry = syncRetry;
+    }
+
+    public ConcurrentHashMap<String, Map<String, Object>> getAssignmentVersions() {
+        return assignmentVersions;
+    }
+
+    public void setAssignmentVersions(Map<String, Map<String, Object>> assignmentVersions) {
+        this.assignmentVersions.clear();
+        this.assignmentVersions.putAll(assignmentVersions);
```

Comment:
Here too this operation should be atomic and I would rather see assignmentVersions be an `AtomicReference<Map<String, Map<String, Object>>>`